src/sidebar/util/time.js

```diff
@@ -195,48 +198,49 @@ function nextFuzzyUpdate(date, now) {
  * update frequency depends on the age of a timestamp.
  *
  * @param {String} date - An ISO 8601 date string timestamp to format.
- * @param {Date} callback - A callback function to call when the timestamp changes.
+ * @param {Function} callback - A callback function to call when the timestamp changes.
  * @return {Function} A function that cancels the automatic refresh.
  */
 function decayingInterval(date, callback) {
   let timer;
   const timeStamp = date ? new Date(date) : null;
-  const update = function() {
+
+  const update = () => {
     const fuzzyUpdate = nextFuzzyUpdate(timeStamp, new Date());
```

Comment:
Thinking more on this, setInterval wont allow for changing the interval time, so the only way I can think of doing this w/o ballooning the stack is to fix a timeout interval and internally do a second check that checks the current time against the fuzzyUpdate + 500 value.

So an aggressive version would be to call this ever 10ms and the check would fail  most of the time, but could probably be relaxed to 500ms. It seems like a huge waste of cpu but games do this sort of thing all the time with timers and frame counts, etc.

I'm not suggesting we make this change, but it might good to investigate how big the stack could get in worse case scenario.  I also would be surprised if there was not a third party function that did this sorta thing.

Reply:
@lyzadanger shelved this to another ticket, no changes required for this PR.

https://github.com/hypothesis/client/issues/1285

Reply:
> I'm new to this code, but does this call itself a few times at most or many many times? My point being that if it's the latter, then setInterval maybe more appropriate so it does not blow up the stack frame.

This function is called at progressively larger intervals. The idea is that when a time is recent, a relative timestamp needs to be updated often ("30 secs ago", "1 minute ago", "2 minutes ago"). As the timestamp gets older, it changes less frequently ("1 Jul" changes once a day) so the interval is adjusted accordingly.

> Upon debugging this with several new replies, I can see the callstack growing with multiple setTimeout functions. 

Can you explain/show what you mean (eg. with a screenshot) here? The recursive call to `update` happens in a separate turn of the event loop so it shouldn't increase the stack size (because the outer `update` is no longer on the stack at that point).